packages/react/collection/src/collection.tsx

```diff
@@ -22,9 +22,8 @@ function createCollection<E extends React.ElementRef<any> = void, S = {}>(name:
     ssrSyncUseCollectionItemCountRef: React.MutableRefObject<number>;
   };
 
-  const [CollectionContext, useCollectionContext] = createContext<CollectionContextValue>(
-    name + 'CollectionContext',
-    providerName
+  const [CollectionProviderImpl, useCollectionContext] = createContextObj<CollectionContextValue>(
```

Comment:
I wasn't sure what to name the `CollectionProviderImpl` part here because we already have a `CollectionProvider` in this file that wraps it.

Reply:
Yeah I guess it makes sense. I've been thinking about it while reviewing these. I think the reason is because we've dropped the `Context` part in the name. When something was before `AccordionContext` and then used `<AccordionContext.Provider>`, that would translate now to `AccordionContextProvider` and here `CollectionContextProvider`.

So maybe we just need to make them all more explicit like that?

But at the same time feels kinda annoying… cause a provider is obviously a context thing…

Reply:
Haha yeah I dropped the `Context` part from the name because we don't do `SSRContextProvider` for example, it's just `SSRProvider`.

Are you happy for me to leave as is?

Reply:
Yeah I'm happy with this.